sky/utils/command_runner.py

```diff
@@ -104,13 +104,22 @@ def ssh_options_list(
     }
     # SSH Control will have a severe delay when using docker_ssh_proxy_command.
     # TODO(tian): Investigate why.
+    #
+    # We disable ControlMaster when ssh_proxy_command is used, because the
+    # master connection will be idle although the connection might be shared
+    # by other ssh commands that is not idle. In that case, user's custom proxy
+    # command may drop the connection due to idle timeout, since it will only
+    # see the idle master connection. It is an issue even with the
+    # ServerAliveInterval set, since the keepalive message may not be recognized
+    # by the custom proxy command, such as AWS SSM Session Manager.
+    #
     # We also do not use ControlMaster when we use `kubectl port-forward`
     # to access Kubernetes pods over SSH+Proxycommand. This is because the
     # process running ProxyCommand is kept running as long as the ssh session
     # is running and the ControlMaster keeps the session, which results in
     # 'ControlPersist' number of seconds delay per ssh commands ran.
     if (ssh_control_name is not None and docker_ssh_proxy_command is None and
-            not disable_control_master):
+            ssh_proxy_command is None and not disable_control_master):
```

Comment:
Curious, how much of a performance regression is this? I'm wondering if there are any recommendations we can make for users to configure large timeouts for SSM.

Reply:
The max SSM timeout is 60 minutes, and it may still not enough if a user tries to keep `sky logs` the output. https://docs.aws.amazon.com/systems-manager/latest/userguide/session-preferences-timeout.html

It seems to me that `ssh_proxy_command` is very customizable, so we have to be conservative here and accept the overhead for disabling the control master.

I was trying to use proxy command for the master branch, but it seems we actually cannot even work well with a normal ssh proxy:
1. `sky launch -c jump --cloud aws --region us-east-1`
2. Set the proxy command:
```yaml
aws:
  ssh_proxy_command: ssh -W %h:%p -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null jump
```
3. `sky launch -c with-jump --cpus 2 echo hi`: this fails to submit jobs with 255 returncode (disconnected), but correctly with the current branch.